dotCMS/src/main/java/com/dotcms/rendering/velocity/services/TemplateLoader.java

```diff
@@ -0,0 +1,102 @@
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+
+import java.io.InputStream;
+
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+/**
+ * @author will
+ *
+ *         To change this generated comment edit the template variable "typecomment":
+ *         Window>Preferences>Java>Templates. To enable and disable the creation of type comments go
+ *         to Window>Preferences>Java>Code Generation.
+ */
+public class TemplateLoader implements DotLoader {
+
+
+
+    public InputStream buildVelocity(Template template, PageMode mode, String filePath)
+            throws DotStateException, DotDataException {
+        Identifier identifier = APILocator.getIdentifierAPI()
+            .find(template);
+        return buildVelocity(template, identifier, mode, filePath);
+    }
+
+    public InputStream buildVelocity(Template template, Identifier identifier, PageMode mode, String filePath) {
+
+
+        StringBuilder templateBody = new StringBuilder();
+
+
+        templateBody.append(Constants.TEMPLATE_PREPROCESS);
+        templateBody.append(template.getBody());
+        templateBody.append(Constants.TEMPLATE_POSTPROCESS);
+
+
+
+        return writeOutVelocity(filePath, templateBody.toString());
+    }
+
+
+
+    @Override
+    public InputStream writeObject(String id1, String id2, PageMode mode, String language, String filePath)
+            throws DotDataException, DotSecurityException {
+
+        Identifier identifier = APILocator.getIdentifierAPI()
+            .find(id1);
+        VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+        Template template = null;
+        if (mode.showLive) {
+            template = (Template) versionableAPI.findLiveVersion(identifier, sysUser(), true);
+
+        } else {
+            template = (Template) versionableAPI.findWorkingVersion(identifier, sysUser(), true);
+        }
+
+        Logger.debug(this, "DotResourceLoader:\tWriting out Template inode = " + template.getInode());
+
+        return buildVelocity(template, mode, filePath);
+
+
+    }
+
+    @Override
+    public void invalidate(Object obj) {
+        for (PageMode mode : PageMode.values()) {
+            invalidate(obj, mode);
+        }
+
+    }
+
+    @Override
+    public void invalidate(Object obj, PageMode mode) {
+        Template template = (Template) obj;
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        velocityRootPath += java.io.File.separator;
+
+        String folderPath = mode.name() + java.io.File.separator;
+        String filePath = folderPath + template.getIdentifier() + "." + VelocityType.TEMPLATE.fileExtension;
+        java.io.File f = new java.io.File(velocityRootPath + filePath);
+        f.delete(); // todo: check if the file exists before remove?
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)